net/net_tcp.c

```diff
@@ -248,9 +248,12 @@ int tcp_init_sock_opt(int s, const struct tcp_conn_profile *prof, enum si_flags
 #endif
 	/* tos*/
 	optval = tos;
-	if (setsockopt(s, IPPROTO_IP, IP_TOS, (void*)&optval,sizeof(optval)) ==-1){
-		LM_WARN("setsockopt tos: %s\n",	strerror(errno));
-		/* continue since this is not critical */
+	if (sock_tos > 0) optval = sock_tos;
```

Comment:
Could you do here a:
```
optval = (sock_tos > 0) ? sock_tos : tos ;
```
It's more clear and compact :)

Reply:
fixed, thank you